internal/attribute/attribute.go

```diff
@@ -49,26 +49,24 @@ func AsBoolSlice(v interface{}) []bool {
 	if rv.Type().Kind() != reflect.Array {
 		return nil
 	}
-	var zero bool
-	correctLen := rv.Len()
-	correctType := reflect.ArrayOf(correctLen, reflect.TypeOf(zero))
-	cpy := reflect.New(correctType)
-	_ = reflect.Copy(cpy.Elem(), rv)
-	return cpy.Elem().Slice(0, correctLen).Interface().([]bool)
+	cpy := make([]bool, rv.Len())
+	if len(cpy) > 0 {
+		_ = reflect.Copy(reflect.ValueOf(cpy), rv)
+	}
+	return cpy
 }
 
 // AsInt64Slice converts an int64 array into a slice into with same elements as array.
 func AsInt64Slice(v interface{}) []int64 {
 	rv := reflect.ValueOf(v)
-	if rv.Type().Kind() != reflect.Array {
+	if rv.Type().Kind() != reflect.Array || rv.Len() == 0 {
```

Comment:
Why is this different than all the other methods? Shouldn't this be the preferred approach so we do not allocate `cpy` below?

Reply:
Good spot, I was tired and missed this my bad.
```suggestion
	if rv.Type().Kind() != reflect.Array {
```

Reply:
It would be a BC break as we used to always return an empty slice and not a `nil` slice. 
It would also require us to call `rv.Len()` twice or introduce a variable and `len()` was a tiny little bit faster.